### PR TITLE
Add application for audio retargeting

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
cc @S-Dafarra

### CODE REVIEW

**Review by S-Dafarra — APPROVED** [review]
Great!

File: app/scripts/audioForRetargeting.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  </module>
+
+   <module>
+    <name>yarphear</name>
+    <parameters>--name /virtualizer/yarphear:i</parameters>
+    <node>icub-virtualizer</node>
+  </module>
+
+  <!-- Connections -->
+  <!-- Hand retargeting  -->
```
Comment (kouroshD) [inline app/scripts/audioForRetargeting.xml]: Probably this is a leftover, you can delete this.

**Review by kouroshD — CHANGES_REQUESTED** [review]

**Comment by kouroshD** [comment]
@GiulioRomualdi Thank you for the PR, that's great. Before merging, can you please link to the issue here. It is not clear the functionalities of different options in yarp devices and the functionalities of the modules; or you can add the explanation here.

**Comment by GiulioRomualdi** [comment]
Following @randaz81 suggestions https://github.com/robotology/yarp/issues/2388#issuecomment-713418289 I dropped `yarphear` in favor of `yarpdev --device AudioPlayerWrapper --subdevice portaudioPlayer`.

I've also added a call to the `rpc` port in the `xml` file. Thanks to this we can avoid to open a new terminal to `start` the process. 

In order to use the application we just have to `run all` e then `connect all`

**Comment by S-Dafarra** [comment]
@kouroshD @GiulioRomualdi can we merge this?

**Comment by GiulioRomualdi** [comment]
From my side yes

**Comment by kouroshD** [comment]
@GiulioRomualdi  @S-Dafarra  Please add a documentation to the xml file. At least wringing the functionality of each module, i.e., `AudioRecorderWrapper` and `AudioPlayerWrapper`. If you can write about the parameters it will be great.

**Comment by GiulioRomualdi** [comment]
I don't have any idea of the meaning of some parameters passed to the `yarpdev`. I've just finished the work done by @miladshafiee in the past. 
However, I'm pretty sure that:
- `AudioRecorderWrapper`: it's a device required to record the audio from the microphone
- `--name <port_name>`: the options `--name` is used to set the name of the port
- `AudioPlayerWrapper` it's a device required to hear the audio.

The ports opened by `AudioRecorderWrapper` are connected to the one opened by `AudioPlayerWrapper`

**Review by kouroshD — APPROVED** [review]

**Comment by kouroshD** [comment]
@GiulioRomualdi  Thank you for the explanation, we can go on with the merge.

**Pull request merged**